apps/teamsjs-diagnostic-app/src/apis/AppApi.tsx

```diff
@@ -0,0 +1,21 @@
+import { app, ResumeContext } from '@microsoft/teams-js';
+
+export const registerOnResume = async (): Promise<string> => {
+    console.log("Calling registerOnResume API");
+  
+    try {
+  
+      app.lifecycle.registerOnResumeHandler((context: ResumeContext): void => {
+        console.log('Successfully called with context: ' + JSON.stringify(context));
+        app.notifySuccess();
+      });
+  
+      console.log("Registered on resume handler successfully");
+  
+      return app.Messages.Success;
+    } catch (error: any) {
+      console.error(`Failed to register resume handler: ${error.toString()}`);
+      throw new Error(`Failed to register resume handler: ${error.toString()}`);
```

Comment:
```suggestion
      const registerError = new Error(`Failed to register resume handler: ${error.toString()}`);
      console.error(registerError.message);
      throw registerError;
```

Doing something like this would make it so you only have your error string once and don't have to keep the two strings in sync. Alternatively, you could choose to just re-throw the unaltered error you already received, like this:

```suggestion
      console.error(`Failed to register resume handler: ${error.toString()}`);
      throw error;
```

There are pros and cons to both approaches, so I'll leave it up to you!